### PR TITLE
Move from 'attached' to 'bind' for custom attribute entry point

### PR DESCRIPTION
**Pull request by jfstephe**
Related to #4 I've made the following alterations:

1). Moved common code to base class (no functional change)
2). Used 'bind' instead of 'attached' as entry point
3). Added 'unbind' to unset the interactable

I tested with dropzone and draggable only and code reviewed the rest and I think (!) everything should be ok.

Notes: I also fixed a typo in the test that meant that it wouldn't compile. The test still didn't pass for me after this update (before any PR related changes were made) but I think that's for a later PR from someone :-).

### CODE REVIEW

File: config.js [hunk 1]
```diff
@@ -8,6 +8,7 @@ System.config({
 
   map: {
     "aurelia-polyfills": "npm:aurelia-polyfills@1.1.0",
+    "interact": "npm:interact.js@1.2.6",
     "interact.js": "npm:interact.js@1.2.6",
```
Comment (eriklieben) [inline config.js:12]: we can probably remove this line, will look at that next week as it isn't blocking/ causing issues. Will move it from typings to 2.0 @types.

**Review by eriklieben — APPROVED** [review]

**Comment by eriklieben** [comment]
Thanks for your contribution and helping out on making this plugin great for the community!

Tested it in JSPM skeleton and aurelia-CLI Aurelia app. I will look at the failing tests next week when I have some more time on my hands and fix them, and move the typescript typings to the new @types format.

Just published changes to npm 🎉 

Sorry for the delay in processing it.

**Comment by jfstephe** [comment]
That's awesome. Thanks!

**Comment by jfstephe** [comment]
Thanks for merging and publishing.

I’ve got another one for you now - last one for a while I.... almost.... promise ??.

https://github.com/eriklieben/aurelia-interactjs/pull/7

This also fixes the test.

If you could merge and publish this that’d be awesome.

Thanks,
John



Sent from Mail<https://go.microsoft.com/fwlink/?LinkId=550986> for Windows 10

**Pull request merged**